src/main/java/org/jvnet/hudson/test/JavaNetReverseProxy2.java

```diff
@@ -0,0 +1,99 @@
+package org.jvnet.hudson.test;
+
+import hudson.Util;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jetty.ee9.servlet.ServletContextHandler;
+import org.eclipse.jetty.ee9.servlet.ServletHolder;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.ContextHandlerCollection;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+
+/**
+ * Acts as a reverse proxy, so that during a test we can avoid hitting updates.jenkins.io.
+ *
+ * <p>
+ * The contents are cached locally.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class JavaNetReverseProxy2 extends HttpServlet {
```

Comment:
Not new code, so please only review the _diff_ of this class with the `JavaNetReverseProxy` class (left as an exercise for the reader).